File: runtime-speech/src/commonMain/kotlin/io/github/nikhilbhutani/TranscriptionResult.kt
package io.github.nikhilbhutani

/**
 * Detailed result from speech-to-text transcription.
 */
data class TranscriptionResult(
    /**
     * Full transcribed text.
     */
    val text: String,

    /**
     * Individual segments with timing information.
     */
    val segments: List<Segment>,

    /**
     * Detected or specified language code.
     */
    val language: String,

    /**
     * Total audio duration in milliseconds.
     */
    val durationMs: Long
)

/**
 * A timed segment of transcribed text.
 */
data class Segment(
    /**
     * Transcribed text for this segment.
     */
    val text: String,

    /**
     * Start time in milliseconds from audio start.
     */
    val startMs: Long,

    /**
     * End time in milliseconds from audio start.
     */
    val endMs: Long
)

File: swift/Sources/DeviceAiStt/Models/TranscriptionResult.swift
/// Full transcription result. Mirrors `TranscriptionResult` + `Segment` in kotlin/speech exactly.
public struct TranscriptionResult: Sendable {
    public let text:      String
    public let segments:  [TranscriptionSegment]
    public let language:  String
    public let durationMs: Int64
}

public struct TranscriptionSegment: Sendable {
    public let text:    String
    public let startMs: Int64
    public let endMs:   Int64
}
